Add support for TLS servers

ok-john opened the pull request:
This PR adds support for TLS servers by adding additional command line arguments for `domain`, `cert` and `key` which represent the url, path to certificate and key files respectively.

This PR also adds a build option `make static` which generates all the static assets required to compile the project. This needs to be run before first build and the only location this info exists is in your documentation under contributing.

I tested this on my own servers and it works.

Commit:
Fixed an issue with the install script failing on mac on the checksum verify step.

Commit:
Fixed no newline at end of file

Commit:
refactor installer script to make it more robust and flexible. Adds optional arguments, a path to manual download if the automatic checks fail, and fallbacks to curl if wget is not installed.

Commit:
fixed some typos

Commit:
Add support for TLS servers

sundowndev commented on cmd/serve.go:
```diff
@@ -33,11 +37,19 @@ func init() {
 	cmd.PersistentFlags().StringSliceVar(&opts.EnvFiles, "env-file", []string{}, "Env files to parse environment variables from (looks for .env by default)")
 }
 
+func fmtLetsEncrypt(sitename string) (string, string) {
+	return fmt.Sprintf("/etc/letsencrypt/live/%s/fullchain.pem", sitename),
+		fmt.Sprintf("/etc/letsencrypt/live/%s/privkey.pem", sitename)
+}
+
```
This function is not windows-compatible. Anyway I think it's not worth to guess the cert and key paths, if the user doesn't specify it, the program exit with error.

sundowndev commented on cmd/serve.go:
```diff
@@ -70,6 +82,15 @@ func NewServeCmd(opts *ServeCmdOptions) *cobra.Command {
 				log.Fatal(err)
 			}
 
+			if len(opts.Domain) != 0 {
+				if len(opts.CertfilePath) == 0 || len(opts.KeyfilePath) == 0 {
+					opts.CertfilePath, opts.KeyfilePath = fmtLetsEncrypt(opts.Domain)
+				}
+				if err := srv.ListenAndServeTLS(opts.Domain+":443", opts.CertfilePath, opts.KeyfilePath); err != nil && err != http.ErrServerClosed {
+					log.Fatalf("listen: %s\n", err)
+				}
+			}
+
```
I just noticed we don't have a addr flag to listen to a different address. I think it's worth adding a new `--addr` flag and use it for both TLS and non-TLS.

`--addr` should be empty by default and we should use TLS when `--cert` or `--key` is not empty.

In the [Gin's server implementation](https://github.com/gin-gonic/gin/blob/b04917c53e310e3746ec90cd93106cda8c956211/ginS/gins.go#L137-L146), it's the same option for both methods.

sundowndev reviewed (changes requested):
Thanks for your PR 👍🏻 

I left few comments about the default paths and CLI options. Also could you add a few lines in the docs just to let the users know we support TLS ? See [this file](https://github.com/sundowndev/phoneinfoga/blob/master/docs/getting-started/usage.md#launching-the-web-server).

ok-john commented:
Hey @sundowndev,

Thanks for the comments - at the moment my availability is a bit limited.

Next time I'm free I'll look at getting to those comments, if you or someone else would like to address those changes and push to this PR that's cool too.

Thanks.

dependabot[bot] committed:
fix(deps): bump goreleaser/goreleaser-action from 4.1.1 to 4.2.0

Bumps [goreleaser/goreleaser-action](https://github.com/goreleaser/goreleaser-action) from 4.1.1 to 4.2.0.
- [Release notes](https://github.com/goreleaser/goreleaser-action/releases)
- [Commits](https://github.com/goreleaser/goreleaser-action/compare/v4.1.1...v4.2.0)

---
updated-dependencies:
- dependency-name: goreleaser/goreleaser-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
fix(deps): bump actions/setup-go from 3.2.0 to 3.5.0

Bumps [actions/setup-go](https://github.com/actions/setup-go) from 3.2.0 to 3.5.0.
- [Release notes](https://github.com/actions/setup-go/releases)
- [Commits](https://github.com/actions/setup-go/compare/v3.2.0...v3.5.0)

---
updated-dependencies:
- dependency-name: actions/setup-go
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
fix(deps): bump docker/login-action from 1.14.1 to 2.1.0

Bumps [docker/login-action](https://github.com/docker/login-action) from 1.14.1 to 2.1.0.
- [Release notes](https://github.com/docker/login-action/releases)
- [Commits](https://github.com/docker/login-action/compare/v1.14.1...v2.1.0)

---
updated-dependencies:
- dependency-name: docker/login-action
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
fix(deps): bump actions/setup-python from 4.3.1 to 4.5.0

Bumps [actions/setup-python](https://github.com/actions/setup-python) from 4.3.1 to 4.5.0.
- [Release notes](https://github.com/actions/setup-python/releases)
- [Commits](https://github.com/actions/setup-python/compare/v4.3.1...v4.5.0)

---
updated-dependencies:
- dependency-name: actions/setup-python
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

sundowndev committed:
ci: create homebrew workflow

sundowndev committed:
chore: update makefile

Move golangci-lint install command into install-tools task

dependabot[bot] committed:
fix(deps): bump actions/checkout from 3.0.0 to 3.3.0

Bumps [actions/checkout](https://github.com/actions/checkout) from 3.0.0 to 3.3.0.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v3.0.0...v3.3.0)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Commit:
squashed some bugs, further testing needed

Commit:
fixed a few typos, should be ready for review

Commit:
removed one additional debug item. ready for review

Commit:
fix shellcheck SC2076 (warning): Remove quotes from right-hand side of =~ to match as a regex rather than literally.

Commit:
fixed 2 shellcheck issues:
1. SC2155 (warning): Declare and assign separately to avoid masking return values.
2. SC2061 (warning): Quote the parameter to -name so the shell won't interpret it.

sundowndev committed:
docs: install instructions

sundowndev committed:
docs: contribute

sundowndev committed:
docs: readme

sundowndev committed:
docs: add homebrew installation

sundowndev committed:
docs: contribute

dependabot[bot] committed:
fix(deps): bump actions/checkout from 3.3.0 to 3.4.0

Bumps [actions/checkout](https://github.com/actions/checkout) from 3.3.0 to 3.4.0.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v3.3.0...v3.4.0)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

sundowndev committed:
ci: homebrew workflow

dependabot[bot] committed:
fix(deps): bump actions/checkout from 3.4.0 to 3.5.0

Bumps [actions/checkout](https://github.com/actions/checkout) from 3.4.0 to 3.5.0.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v3.4.0...v3.5.0)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
fix(deps): bump actions/setup-go from 3.5.0 to 4.0.0

Bumps [actions/setup-go](https://github.com/actions/setup-go) from 3.5.0 to 4.0.0.
- [Release notes](https://github.com/actions/setup-go/releases)
- [Commits](https://github.com/actions/setup-go/compare/v3.5.0...v4.0.0)

---
updated-dependencies:
- dependency-name: actions/setup-go
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
fix(deps): bump actions/checkout from 3.5.0 to 3.5.2

Bumps [actions/checkout](https://github.com/actions/checkout) from 3.5.0 to 3.5.2.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v3.5.0...v3.5.2)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Commit:
fix: colored text on windows

Commit:
fix: server logs output

dependabot[bot] committed:
fix(deps): bump actions/setup-python from 4.5.0 to 4.6.0

Bumps [actions/setup-python](https://github.com/actions/setup-python) from 4.5.0 to 4.6.0.
- [Release notes](https://github.com/actions/setup-python/releases)
- [Commits](https://github.com/actions/setup-python/compare/v4.5.0...v4.6.0)

---
updated-dependencies:
- dependency-name: actions/setup-python
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

sundowndev commented:
It's been a year, any news on this @ok-john ?

ok-john commented:
> It's been a year, any news on this @ok-john ?

I could work on this now i have time - is this still a requested feature?

sundowndev commented:
@ok-john At this time it's not requested. If you'd like to request it, please create an issue first then create a PR. If you don't need this feature, then I think we can close this PR.

sam05801601-eng commented on cmd/serve.go:
```diff
@@ -11,14 +15,14 @@ import (
-	"os"
 )
 
 type ServeCmdOptions struct {
```
+919847951352
